### ISSUE TITLE
Support recursive flag in configuration file

### ISSUE THREAD

**Issue by steven-edgar** [terraform-docs/terraform-docs#557]
### What problem are you facing?

I would like to be able to have my auto-generated documention produced recursively for all submodules, which I can achieve using the --recursive flag, however I cannot set that flag in my .terraform-docs.yml file to ensure all those updating the documentation with terraform-docs consistently update recursively.

### How could terraform-docs help solve your problem?

Support the following in .terraform-docs.yml

```yaml
settings:
   recursive: true
```

**Comment by khos2ow** [comment]
The original reason `recursive` wasn't added to config file and is only available via cli flag, was the submodule can provide its own `.terraform-docs.yml` file to override its parent configuration.

```
$ tree .
.
├── README.md
├── main.tf
├── modules
│   ├── submodule-1
│   │   ├── README.md
│   │   ├── main.tf
│   │   ├── .terraform-docs.yml
│   │   ├── variables.tf
│   │   └── versions.tf
│   └── submodule-2
│       ├── README.md
│       ├── main.tf
│       ├── variables.tf
│       └── versions.tf
├── outputs.tf
├── .terraform-docs.yml
├── variables.tf
└── versions.tf
```

Note that `.submodule-1` is going to override `.terraform-docs.yml` config. For that reason having `recursive` as part of the configuration can potentially be confusing.

That said, I'm in the middle of experimenting with recursive config to be able to partially override them (as opposed to now which a child completely overrides its parent's config and cannot reuse anything from there). With that in place I'm confident we can identify who defines `recursive: true` and reject it if a child config attempts to do it.

**Comment by khos2ow** [comment]
This is now implemented in #570 with following config stanza:

```yaml
recursive:
  enabled: false
  path: modules
```
